src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -577,6 +580,39 @@ public boolean hasTag(String tag) {
 				log.printError();
 				return null;
 			}
+			Matcher literalClassInfoMatcher = LITERAL_CLASSINFO_PARSE.matcher(expr);
```

Comment:
could we please put as much of this logic as possible into helper methods so that we don't make these methods even bigger